backend/src/apiserver/server/pipeline_server_test.go

```diff
@@ -281,6 +281,106 @@ func TestListPipelinesPublic(t *testing.T) {
 
 }
 
+func TestGetPipelineByName_OK(t *testing.T) {
+	httpServer := getMockServer(t)
+	// Close the server when test finishes
+	defer httpServer.Close()
+	clientManager := resource.NewFakeClientManagerOrFatal(util.NewFakeTimeForEpoch())
+	resourceManager := resource.NewResourceManager(clientManager)
+	pipelineServer := PipelineServer{resourceManager: resourceManager, httpClient: httpServer.Client(), options: &PipelineServerOptions{CollectMetrics: false}}
+	pipeline, err := pipelineServer.CreatePipeline(context.Background(), &api.CreatePipelineRequest{
+		Pipeline: &api.Pipeline{
+			Url:  &api.Url{PipelineUrl: httpServer.URL + "/arguments-parameters.yaml"},
+			Name: "argument-parameters",
+			ResourceReferences: []*api.ResourceReference{{
+				Key: &api.ResourceKey{
+					Id:   "ns1",
+					Type: api.ResourceType_NAMESPACE,
+				}},
+			},
+		}})
+	assert.Nil(t, err)
+	assert.NotNil(t, pipeline)
+	newPipeline, err := pipelineServer.GetPipelineByName(context.Background(),
+		&api.GetPipelineByNameRequest{
+			Name:      pipeline.Name,
+			Namespace: "ns1",
+		})
+	assert.Nil(t, err)
+	assert.NotNil(t, newPipeline)
+	assert.Equal(t, "argument-parameters", pipeline.Name)
+}
+
+func TestGetPipelineByName_Shared_OK(t *testing.T) {
```

Comment:
@zijianjoy about writing a unit test .. what about this one? It creates a shared pipeline - with no namespace, and then tries to read it. I am not quite sure how to write a unit test that tests the type of installation .. ?

Reply:
Thank you Diana, I think this unit test is good for simulating the shared namespace.